### PR TITLE
Disable fog (opacity = 0) on globe with terrain elevation

### LINKED ISSUE #4928: The unsupported Fog/Horizon is isn't disabled properly on Globe with Terrain3D
Repro here:
- #4926 

Changing the pitch will remove all texture, resulting in a white screen:

https://github.com/user-attachments/assets/25a589f3-a148-4a23-a108-84f977ab0a47

### PR DESCRIPTION
**Pull request by birkskyum**
Closes #4928 
- #4928 

Towards 
- #4927

As mentioned here, the globe doesn't support Fog "calculateFogMatrix is not supported on globe projection":

https://github.com/maplibre/maplibre-gl-js/blob/ef6e9eec447e04ce600b6562d7ad6fddca8a802c/src/geo/projection/globe_transform.ts#L682

Unfortunately, it's still trying to render it, which consistently breaks the map whenever the map is pitched just a bit.

This tiny PR simply set the opacity of the fog/horizon to 0 in globe+terrain mode. When the globe does support fog with terrain, this can easily be removed again:
- #4924 

**Fog is still active on Terrain3D without Globe**

### With Globe - zoom level before mercator transition

Before
Screenshot 2024-11-02 at 17 13 01

After  - **main branch**
Screenshot 2024-11-02 at 17 18 01

After -  **local branch**
Screenshot 2024-11-02 at 17 10 52

### With Globe - high zoom, after transition to mercator
Before
Screenshot 2024-11-02 at 18 30 42

After
Screenshot 2024-11-02 at 18 27 40

### Without Globe - **Fog Still Works**

Before/After
Screenshot 2024-11-02 at 17 10 29

## Launch Checklist

 - [X] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!

### CODE REVIEW

File: src/render/draw_terrain.ts [hunk 1]
```diff
@@ -88,7 +89,7 @@ function drawTerrain(painter: Painter, terrain: Terrain, tiles: Array<Tile>) {
         gl.bindTexture(gl.TEXTURE_2D, texture.texture);
         const eleDelta = terrain.getMeshFrameDelta(tr.zoom);
         const fogMatrix = tr.calculateFogMatrix(tile.tileID.toUnwrapped());
-        const uniformValues = terrainUniformValues(eleDelta, fogMatrix, painter.style.sky, tr.pitch);
+        const uniformValues = terrainUniformValues(eleDelta, fogMatrix, painter.style.sky, tr.pitch, tr instanceof GlobeTransform && tr.isGlobeRendering);
```
Comment (HarelM) [inline src/render/draw_terrain.ts:92]: The idea is not to know what instance the trasform is, it breaks the abstraction principle.
I think you can use the transform's transition or something similar, or check the current style definition.

Comment (birkskyum) [reply src/render/draw_terrain.ts]: Right, I took it from the `painter.style.projection.name === 'globe'`

Comment (birkskyum) [reply src/render/draw_terrain.ts]: Your right, It's better to get it from the painter

Comment (HarelM) [reply src/render/draw_terrain.ts]: You can use getProjectionData and then projectionTransition:
https://github.com/maplibre/maplibre-gl-js/blob/ef6e9eec447e04ce600b6562d7ad6fddca8a802c/src/geo/projection/projection_data.ts#L40

**Comment by HarelM** [comment]
Can you add a unit test to cover this issue somehow?

**Comment by birkskyum** [comment]
@HarelM , I've added a unit test to make sure u_fog_ground_blend_opacity = 0 for globe mode

**Comment by birkskyum** [comment]
@HarelM , I've added a unit test, and a render test.

File: test/integration/render/tests/terrain/fog-sky-blend-globe/bug-when-fog-is-enabled.png
Comment (HarelM) [inline test/integration/render/tests/terrain/fog-sky-blend-globe/bug-when-fog-is-enabled.png:1]: Is this file needed? I don't see a style.json related to it...?

**Comment by HarelM** [comment]
Overall approved, let me know if the file I commented on is needed.

**Comment by birkskyum** [comment]
The file was added to prove that the render test does capture these symptoms - it could have been added with the style before this pr. It's not strictly needed.

File: test/integration/render/tests/terrain/fog-sky-blend-globe/bug-when-fog-is-enabled.png
Comment (birkskyum) [reply test/integration/render/tests/terrain/fog-sky-blend-globe/bug-when-fog-is-enabled.png:1]: The file was added to prove that the render test does capture these symptoms, just for the review - it could have been added with the style before this pr. It's not strictly needed. I have removed it

**Comment by HarelM** [comment]
Please add a changelog entry.

**Review by HarelM — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 ### 🐞 Bug fixes
 - Fix line-placed map-pitch-aligned texts being too large when viewed from some latitudes on a globe ([#4786](https://github.com/maplibre/maplibre-gl-js/issues/4786))
+- Disabled unsupporeted Fog rendering, for Terrain3D on Globe ([#4963](https://github.com/maplibre/maplibre-gl-js/pull/4963))
```
Comment (HarelM) [inline CHANGELOG.md:13]:
```suggestion
- Disabled unsupported Fog rendering, for Terrain3D on Globe ([#4963](https://github.com/maplibre/maplibre-gl-js/pull/4963))
```
